src/NuGet.Core/NuGet.PackageManagement/Utility/XmlUtility.cs

```diff
@@ -1,70 +1,47 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.IO;
 using System.Xml;
 using System.Xml.Linq;
+using XmlUtility = NuGet.Shared.XmlUtility;
 
 namespace NuGet.ProjectManagement
 {
     public static class XmlUtility
     {
+        [Obsolete("This method is obsolete and will be removed in a future release.")]
```

Comment:
Obsolete the class too?

Reply:
Thanks, Nikolche. I just found a typo in the `obsolete` message. I agree with Andy's [comment](https://github.com/NuGet/Home/issues/9091#issuecomment-678644008) on the issue to consider deleting obsolete API in NuGet 6.x.

Reply:
This class has `GetOrCreateDocument` method which requires further refactoring as it accepts additional parameter `INuGetProjectContext` and has a caller from another assembly. As per Andy's feedback on the issue, I tried to avoid any changes to the public API. I will consider refactoring this code in the next PR.